### PR TITLE
[WIP/RFC] use True as default for --cov-config

### PR DESCRIPTION
**Pull request by blueyed**
TODO:

- [ ] refactor to be used with other controllers

### CODE REVIEW

**Comment by blueyed** [comment]
@ionelmc 
Any input on this?
/cc @nedbat (https://github.com/nedbat/coveragepy/issues/788)

File: src/pytest_cov/engine.py [hunk 1]
```diff
@@ -151,10 +150,20 @@ def start(self):
         self.cov = coverage.Coverage(source=self.cov_source,
                                      branch=self.cov_branch,
                                      config_file=self.cov_config)
+
+        # Get absolute location of used config file.
+        parsed_config_file = self.cov.config.config_file
+        if parsed_config_file:
+            parsed_config_file = os.path.abspath(parsed_config_file)
+            assert os.path.exists(parsed_config_file)
```
Comment (ionelmc) [inline src/pytest_cov/engine.py]: I think this would cause failures when coverage config is completely missing. I suspect this is hosing up all the tests.

Comment (blueyed) [reply src/pytest_cov/engine.py]: Tests are mostly broken because it requires coveragepy 5.

**Comment by nedbat** [comment]
@blueyed I took a stab at the coverage.py change: https://github.com/nedbat/coveragepy/tree/nedbat/configfile-simplification

Running the pytest-cov tests with that, there are many failures because the config file handling has to change here.  I'm not sure it's worth it now, but maybe you want to take a stab at it?

**Pull request closed without merging**